.github/workflows/node-release.yml

```
@@ -1,53 +1,41 @@
 name: node-release
 
 on:
+  push:
+    branches: [main]
   workflow_dispatch:
-    inputs:
-      version:
-        description: Choose a release type
-        required: true
-        type: choice
-        options:
-          - prerelease
-          - prepatch
-          - preminor
-          - premajor
-          - patch
-          - minor
-          - major
 
 jobs:
-  bump_version:
-    runs-on: ubuntu-20.04
-
+  release-check:
+    name: Check if version is published
+    runs-on: ubuntu-latest
     defaults:
       run:
-        working-directory: ./
         shell: bash
-
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-          token: ${{ secrets.NODE_PRE_GYP_GITHUB_TOKEN }}
 
-      - name: Setup node
+      - name: Use Node.js from nvmrc
         uses: actions/setup-node@v4
         with:
-          node-version: 18
+          node-version-file: '.nvmrc'
 
-      - name: npm version
+      - name: Check if version is published
+        id: check
         run: |
-          echo "NPM_VERSION=$(npm version ${{ github.event.inputs.version }} --preid pre --no-git-tag-version)" >> "$GITHUB_ENV"
-
-      - name: Push version to git
-        run: |
-          git config user.name github-actions
-          git config user.email github-actions@github.com
-          git commit -a -m "Update node version to ${{ env.NPM_VERSION }} (${{ github.event.inputs.version }})"
-          git push
+          currentVersion="$( node -e "console.log(require('./package.json').version)" )"
+          isPublished="$( npm view @maplibre/maplibre-gl-native versions --json | jq -c --arg cv $currentVersion 'any(. == $cv)' )"
```

I ran `pre-commit run actionlint --all-files` to confirm, this is the last thing!

```suggestion
          isPublished="$( npm view @maplibre/maplibre-gl-native versions --json | jq -c --arg cv "$currentVersion" 'any(. == "$cv")' )"
```

---

It doesn't work when I quote the "$cv" (I think because it is a variable passed from jq?), but just quoting the "$currentVersion" seemed to make the pre-commit ci happy and this check actually still works.

Skipped because published
https://github.com/WifiDB/maplibre-native/actions/runs/7391761598

New release publish
https://github.com/WifiDB/maplibre-native/actions/runs/7391779968